pkg/controller/machinepool/awsactuator_test.go

```diff
@@ -567,10 +571,116 @@ func TestAWSActuator(t *testing.T) {
+					Name:   "tag:Name",
+					Values: []string{"foo-12345-worker-sg"},
+				},
+			},
+			expectedSGIDs: []string{"sg-one", "sg-two"},
+		}, {
+			name:              "ExtraWorkerSecurityGroup and AdditionalSecurityGroupIDs are mutually exclusive",
+			clusterDeployment: testClusterDeployment(),
+			machinePool: func() *hivev1.MachinePool {
+				pool := testMachinePool()
+				pool.Annotations = map[string]string{
+					constants.ExtraWorkerSecurityGroupAnnotation: "extra-security-group",
+				}
+				pool.Spec.Platform.AWS.AdditionalSecurityGroupIDs = []string{"additional-security-group"}
+				return pool
+			}(),
+			masterMachine: testMachine("master0", "master"),
+			expectedCondition: &hivev1.MachinePoolCondition{
+				Type:   hivev1.UnsupportedConfigurationMachinePoolCondition,
+				Status: corev1.ConditionTrue,
+				Reason: "SecurityGroupOptionConflict",
+			},
+		},
 	}
 
-	for _, test := range tests {
+	validateAWSMachineSets := func(t *testing.T, mSets []*machineapi.MachineSet, test ttype) {
+		assert.Equal(t, len(test.expectedMachineSetReplicas), len(mSets), "different number of machine sets generated than expected")
+
+		for _, ms := range mSets {
+			expectedReplicas, ok := test.expectedMachineSetReplicas[ms.Name]
+			if assert.True(t, ok, "unexpected machine set", ms.Name) {
+				assert.Equal(t, expectedReplicas, int64(*ms.Spec.Replicas), "replica mismatch", ms.Name)
+			}
+
+			awsProvider, ok := ms.Spec.Template.Spec.ProviderSpec.Value.Object.(*machineapi.AWSMachineProviderConfig)
+			assert.True(t, ok, "failed to convert to AWSMachineProviderConfig")
+
+			assert.Equal(t, testInstanceType, awsProvider.InstanceType, "unexpected instance type")
+
+			if assert.NotNil(t, awsProvider.AMI, "missing AMI") {
+				if test.expectedAMI.ID != nil {
+					assert.Equal(t, *test.expectedAMI.ID, *awsProvider.AMI.ID, "unexpected AMI ID")
+				}
+				if len(test.expectedAMI.Filters) > 0 {
+					assert.Equal(t, test.expectedAMI.Filters, awsProvider.AMI.Filters, "unexpected AMI filter")
+				}
+			}
+
+			assert.Equal(t, test.expectedKMSKey, *awsProvider.BlockDevices[0].EBS.KMSKey.ARN)
+
+			if test.expectedSubnetIDInMachineSet {
+				providerConfig := ms.Spec.Template.Spec.ProviderSpec.Value.Object.(*machineapi.AWSMachineProviderConfig)
+				assert.NotNil(t, providerConfig.Subnet.ID, "missing subnet ID")
+				assert.Equal(t, "subnet-"+providerConfig.Placement.AvailabilityZone, *providerConfig.Subnet.ID, "unexpected subnet ID")
+			}
+			if test.expectedSGFilters != nil {
+				assert.Equal(t, awsProvider.SecurityGroups[0].Filters, test.expectedSGFilters, "unexpected security group filters")
+			}
+
+			if test.expectedSGIDs != nil {
+				if assert.Equal(t, len(test.expectedSGIDs), len(awsProvider.SecurityGroups)-1, "expected n-1 SecurityGroups holding IDs") {
+					for i := 0; i < len(test.expectedSGIDs); i++ {
+						assert.Equal(t, test.expectedSGIDs[i], *awsProvider.SecurityGroups[i+1].ID, "mismatched security group ID")
+					}
+				}
+			}
```

Comment:
...here. Other than renaming vars, this is the only delta to the func: new logic to validate security group IDs.